tests/test_agent.py

```diff
@@ -251,3 +251,16 @@ def test_agentset_select_by_type():
     # Test with no type specified (should select all agents)
     all_agents = agentset.select()
     assert len(all_agents) == len(mixed_agents)
+
+
+def test_agentset_shuffle():
+    model = Model()
+    test_agents = [TestAgent(model.next_id(), model) for _ in range(4)]
```

Comment:
Maybe add a few more agents to minimize the chance that the orders are accidentally the same.

Reply:
Forgot this was my branch, can do this myself!